idem_provider_azurerm/states/azurerm/key_vault/vault.py

```diff
@@ -0,0 +1,428 @@
+        the Key Vault service may initiate a hard, irrecoverable deletion. The setting is effective only if soft
+        delete is also enabled. Enabling this functionality is irreversible - that is, the property does not accept
+        false as its value.
+
+    :param enabled_for_deployment: A boolean value specifying whether Azure Virtual Machines are permitted to
+        retrieve certificates stored as secrets from the key vault. Defaults to False.
+
+    :param enabled_for_disk_encryption: A boolean value specifying whether Azure Disk Encrpytion is permitted to
+        retrieve secrets from the vault and unwrap keys. Defaults to False.
+
+    :param enabled_for_template_deployment: A boolean value specifying whether Azure Resource Manager is permitted
+        to retrieve secrets from the key vault. Defaults to False.
+
+    :param tags: A dictionary of strings can be passed as tag metadata to the key vault.
+
+    :param connection_auth: A dict with subscription and authentication parameters to be used in connecting to the
+        Azure Resource Manager API.
+
+    Example usage:
+
+    .. code-block:: yaml
+
+        Ensure key vault exists:
+            azurerm.key_vault.vault.present:
+                - name: my_vault
+                - resource_group: my_rg
+                - location: my_location
+                - tenant_id: my_tenant
+                - sku: my_sku
+                - access_policies:
+                  - tenant_id: my_tenant
+                    object_id: my_object
+                    permissions:
+                      keys:
+                        - perm1
+                        - perm2
+                        - perm3
+                      secrets:
+                        - perm1
+                        - perm2
+                        - perm3
+                      certificates:
+                        - perm1
+                        - perm2
+                        - perm3
+                - tags:
+                    contact_name: Elmer Fudd Gantry
+                - connection_auth: {{ profile }}
+    
+    '''
+    ret = {
+        'name': name,
+        'result': False,
+        'comment': '',
+        'changes': {}
+    }
+
+    if not isinstance(connection_auth, dict):
+        ret['comment'] = 'Connection information must be specified via connection_auth dictionary!'
+        return ret
+
+    vault = await hub.exec.azurerm.key_vault.vault.get(
+        name,
+        resource_group,
+        azurearm_log_level='info',
+        **connection_auth
+    )
+
+    if 'error' not in vault:
+        tag_changes = await hub.exec.utils.dictdiffer.deep_diff(vault.get('tags', {}), tags or {})
+        if tag_changes:
+            ret['changes']['tags'] = tag_changes
+    
+        # Checks for changes in the account_policies parameter
+        if len(access_policies) == len(vault.get('properties').get('access_policies', [])):
```

Comment:
if you don't pass in `access_policies`, it defaults to `None` and will throw an error while trying to get `len()`. you probably want to get `len(access_policies or [])`